src/Driver/SQLite3/Driver.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Doctrine\DBAL\Driver\SQLite3;
+
+use Doctrine\DBAL\Driver\AbstractSQLiteDriver;
+use Doctrine\DBAL\Driver\API\SQLite\UserDefinedFunctions;
+use SQLite3;
+
+final class Driver extends AbstractSQLiteDriver
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function connect(array $params): Connection
+    {
+        if (isset($params['path'])) {
+            $filename = $params['path'];
+        } elseif (isset($params['memory'])) {
+            $filename = ':memory:';
+        } else {
```

Comment:
Instead of having the precedence of `path` and `memory`, why not make them mutually exclusive in the new driver? Specifying them both could be deprecated in the old one separately.

Reply:
Sounds good.